src/main/java/com/radixdlt/atomos/ParticleDefinition.java

```diff
@@ -58,7 +63,88 @@
 		return rriMapper;
 	}
 
+	Function<T, Spin> getVirtualizeSpin() {
+		return virtualizeSpin;
+	}
+
 	public boolean allowsTransitionsFromOutsideScrypts() {
 		return allowsTransitionsFromOutsideScrypts;
 	}
+
+	/**
+	 * Creates a Builder for {@link ParticleDefinition}s.
+	 * @param <T> The type of the {@link ParticleDefinition}
+	 * @return The builder
+	 */
+	public static <T extends Particle> Builder<T> builder() {
+		return new Builder<>();
+	}
+
+	/**
+	 * A Builder for {@link ParticleDefinition}s
+	 * @param <T> The type of the particle to be defined
+	 */
+	public static class Builder<T extends Particle> {
+		private Function<T, Set<RadixAddress>> addressMapper;
+		private Function<T, Result> staticValidation = x -> Result.success();
+		private Function<T, RRI> rriMapper;
+		private Function<T, Spin> virtualizeSpin;
+		private boolean allowsTransitionsFromOutsideScrypts;
```

Comment:
Maybe should be an explicit default to `false` here, rather than relying on the default initialisation.

Reply:
Sure can do